api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -173,3 +175,11 @@ type NUMAResourcesOperatorList struct {
 func init() {
 	SchemeBuilder.Register(&NUMAResourcesOperator{}, &NUMAResourcesOperatorList{})
 }
+
+func (ngc *NodeGroupConfig) ToString() string {
```

Comment:
`ToString` is the variant we (mostly me) invented when we want a string serialization but we don't feel ready, or we prefer to, not implement the [Stringer interface](https://go.dev/tour/methods/17)

So in general let's be conscious if our stringification functions are ready or not to be promoted to full Stringer interface. In this case, do you want or prefer to do so?

Reply:
Thanks for this.
I saw the need for this mainly because https://github.com/openshift-kni/numaresources-operator/blob/main/controllers/numaresourcesoperator_controller.go#L359 doesn't print readable value (address).
 I don't see any harm in making this the official stringer implementation for NodeGroupConfig as long we agree on the content presentation:
1. completely nil/empty struct is printed as empty  string ""
2. otherwise, nil fields are filled with the default and all are printed out.
The only concern would be maintenance, e.g adjusting the method to reflect all fields, but this should be part of every update done on the config struct.wdyt?

Reply:
I think maintaining is fine. We can add tests to enforce this, perhaps later:
1. add test(s) which uses String() with non-defaults values
2. add test(s) which do a round trip: initialize object, call String() on it, pass to a function which create an object from its string representation, check it is equal to the initial object (roundtrip test)